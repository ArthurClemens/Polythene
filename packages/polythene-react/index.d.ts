export { Button } from "polythene-react-button";
export { ButtonGroup } from "polythene-react-button-group";
export { Card } from "polythene-react-card";
export { Checkbox } from "polythene-react-checkbox";
export { Dialog } from "polythene-react-dialog";
export { DialogPane } from "polythene-react-dialog-pane";
export { Drawer } from "polythene-react-drawer";
export { FAB } from "polythene-react-fab";
export { Icon } from "polythene-react-icon";
export { IconButton } from "polythene-react-icon-button";
export { IOSSpinner } from "polythene-react-ios-spinner";
export { List } from "polythene-react-list";
export { ListTile } from "polythene-react-list-tile";
export { MaterialDesignProgressSpinner } from "polythene-react-material-design-progress-spinner";
export { MaterialDesignSpinner } from "polythene-react-material-design-spinner";
export { Menu } from "polythene-react-menu";
export { Notification } from "polythene-react-notification";
export { RadioButton } from "polythene-react-radio-button";
export { RadioGroup } from "polythene-react-radio-group";
export { Ripple } from "polythene-react-ripple";
export { Search } from "polythene-react-search";
export { Shadow } from "polythene-react-shadow";
export { Slider } from "polythene-react-slider";
export { Snackbar } from "polythene-react-snackbar";
export { SVG } from "polythene-react-svg";
export { Switch } from "polythene-react-switch";
export { Tabs } from "polythene-react-tabs";
export { TextField, onChangeTextFieldState } from "polythene-react-textfield";
export { Toolbar, ToolbarTitle } from "polythene-react-toolbar";
