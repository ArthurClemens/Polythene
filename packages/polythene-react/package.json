{
  "name": "polythene-react",
  "version": "1.3.3",
  "description": "",
  "main": "dist/polythene-react",
  "module": "dist/polythene-react.mjs",
  "scripts": {
    "clean": "rm -rf dist/*",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/bundling/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/bundling/rollup.es.js",
    "build-standalone": "ENTRY=scripts/standalone.js DEST=./dist/polythene-react-standalone ../../node_modules/rollup/bin/rollup -c ../../scripts/bundling/rollup.browser.js",
    "build": "npm run clean && npm run rollup && npm run build-standalone"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-core-css": "^1.3.3",
    "polythene-core": "^1.3.3",
    "polythene-css": "^1.3.3",
    "polythene-react-base": "^1.3.3",
    "polythene-react-button-group": "^1.3.3",
    "polythene-react-button": "^1.3.3",
    "polythene-react-card": "^1.3.3",
    "polythene-react-checkbox": "^1.3.3",
    "polythene-react-dialog-pane": "^1.3.3",
    "polythene-react-dialog": "^1.3.3",
    "polythene-react-drawer": "^1.3.3",
    "polythene-react-fab": "^1.3.3",
    "polythene-react-icon-button": "^1.3.3",
    "polythene-react-icon": "^1.3.3",
    "polythene-react-ios-spinner": "^1.3.3",
    "polythene-react-list-tile": "^1.3.3",
    "polythene-react-list": "^1.3.3",
    "polythene-react-material-design-progress-spinner": "^1.3.3",
    "polythene-react-material-design-spinner": "^1.3.3",
    "polythene-react-menu": "^1.3.3",
    "polythene-react-notification": "^1.3.3",
    "polythene-react-radio-button": "^1.3.3",
    "polythene-react-radio-group": "^1.3.3",
    "polythene-react-raised-button": "^1.3.3",
    "polythene-react-ripple": "^1.3.3",
    "polythene-react-search": "^1.3.3",
    "polythene-react-shadow": "^1.3.3",
    "polythene-react-slider": "^1.3.3",
    "polythene-react-snackbar": "^1.3.3",
    "polythene-react-svg": "^1.3.3",
    "polythene-react-switch": "^1.3.3",
    "polythene-react-tabs": "^1.3.3",
    "polythene-react-textfield": "^1.3.3",
    "polythene-react-toolbar": "^1.3.3"
  },
  "peerDependencies": {
    "react": "^16.2.0",
    "react-hyperscript": "^3.0.0"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://arthurclemens.com)",
  "homepage": "https://github.com/ArthurClemens/polythene",
  "license": "MIT"
}
