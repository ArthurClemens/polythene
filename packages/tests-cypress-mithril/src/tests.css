.page__test__component .align-items-center {
  display: flex;
  align-items: center;
}