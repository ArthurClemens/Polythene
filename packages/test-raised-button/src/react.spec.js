import { runSnapshots } from "../../../scripts/tests/react-snapshots";
import { renderer, keys, Button } from "polythene-react";
import specTests from "./spec-tests.js";
import reactTests from "./tests-react.js";

runSnapshots({
  tests: specTests({ Button, renderer, keys }).concat(reactTests),
  renderer
});
