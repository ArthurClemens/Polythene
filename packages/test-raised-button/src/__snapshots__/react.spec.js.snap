// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button row 1`] = `
<div
  className="pe-button-row"
>
  <RaisedButton
    key="one"
    label="One"
  />
  <RaisedButton
    key="two"
    label="Two"
  />
  <RaisedButton
    key="three"
    label="Three"
  />
</div>
`;

exports[`Dark tone class + light tone 1`] = `
<div
  className="pe-button-row"
  style={
    Object {
      "background": "#fff",
    }
  }
>
  <RaisedButton
    label="Normal"
    tone="light"
  />
  <RaisedButton
    disabled={true}
    label="Disabled"
    tone="light"
  />
  <RaisedButton
    className="tests-raised-button-themed-button"
    label="Theme"
    tone="light"
  />
</div>
`;

exports[`Dark tone class + light tone class 1`] = `
<div
  className="pe-button-row pe-light-tone"
  style={
    Object {
      "background": "#fff",
    }
  }
>
  <RaisedButton
    label="Normal"
  />
  <RaisedButton
    disabled={true}
    label="Disabled"
  />
  <RaisedButton
    className="tests-raised-button-themed-button"
    label="Theme"
  />
</div>
`;

exports[`No options 1`] = `
<Button
  animateOnTap={false}
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: animateOnTap (false) 1`] = `
<Button
  animateOnTap={false}
  label="Don't animate shadow"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: children 1`] = `
<Button
  animateOnTap={false}
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
>
  <span>
    Child
  </span>
</Button>
`;

exports[`Option: class 1`] = `
<Button
  animateOnTap={false}
  className="class-x"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: className 1`] = `
<Button
  animateOnTap={false}
  className="className-x"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: content 1`] = `
<Button
  animateOnTap={false}
  content={
    <span>
      Content
    </span>
  }
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: disabled (false) 1`] = `
<Button
  animateOnTap={false}
  disabled={false}
  label="Not disabled"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: disabled (true) 1`] = `
<Button
  animateOnTap={false}
  disabled={true}
  label="Disabled"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={0}
    />
  }
/>
`;

exports[`Option: disabled -- dark tone class 1`] = `
<Button
  animateOnTap={false}
  disabled={true}
  label="Label"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={0}
    />
  }
/>
`;

exports[`Option: element 1`] = `
<Button
  animateOnTap={false}
  element="button"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: id 1`] = `
<Button
  animateOnTap={false}
  id="id-x"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: inactivate (2s) (JSX) 1`] = `
<RaisedButton
  inactivate={2}
  label="Inactivated for 2s"
/>
`;

exports[`Option: inactivate (2s) 1`] = `
<Button
  animateOnTap={false}
  inactivate={2}
  label="Inactivated for 2s"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: inactive (false) 1`] = `
<Button
  animateOnTap={false}
  inactive={false}
  label="Not inactive"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: inactive (true) 1`] = `
<Button
  animateOnTap={false}
  inactive={true}
  label="Inactive"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: ink (false) 1`] = `
<Button
  animateOnTap={false}
  ink={false}
  label="No ink"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: label -- dark tone class (should be app's primary color) 1`] = `
<Button
  animateOnTap={false}
  label="Label"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: label 1`] = `
<Button
  animateOnTap={false}
  label="Label"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: raised (with option z: 2) (JSX) 1`] = `
<RaisedButton
  label="Button"
  z={2}
/>
`;

exports[`Option: raised (with option z: 2) 1`] = `
<Button
  animateOnTap={false}
  label="Raised to 2"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
  z={2}
/>
`;

exports[`Option: raised (with option z: 5) 1`] = `
<Button
  animateOnTap={false}
  label="Raised to 5"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
  z={5}
/>
`;

exports[`Option: selected 1`] = `
<Button
  animateOnTap={false}
  label="Selected"
  parentClassName="pe-button pe-text-button pe-raised-button"
  selected={true}
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Option: style (colors) 1`] = `
<Button
  animateOnTap={false}
  label="Styled"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
  style={
    Object {
      "backgroundColor": "#EF6C00",
      "color": "#fff",
    }
  }
/>
`;

exports[`Option: tabindex 1`] = `
<Button
  animateOnTap={false}
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
  tabIndex={3}
/>
`;

exports[`Option: wash (false) 1`] = `
<Button
  animateOnTap={false}
  label="No wash"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
  wash={false}
/>
`;

exports[`Themed button (hover color) 1`] = `
<Button
  animateOnTap={false}
  className="tests-raised-button-hover-button"
  label="Hover button"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Themed button (should be red) 1`] = `
<Button
  animateOnTap={false}
  className="tests-raised-button-themed-button"
  label="Themed button"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={[Function]}
    />
  }
/>
`;

exports[`Themed button (with option disabled) 1`] = `
<Button
  animateOnTap={false}
  className="tests-raised-button-themed-button"
  disabled={true}
  label="Disabled themed button"
  parentClassName="pe-button pe-text-button pe-raised-button"
  shadowComponent={
    <Shadow
      animated={true}
      z={0}
    />
  }
/>
`;

exports[`With router 1`] = `
<Route
  render={[Function]}
/>
`;
